### PR TITLE
Don't allow removal of nodes with edges

### PR DESCRIPTION
**Pull request by younik**
## Description
 

This PR implements the first solution described in https://github.com/GFNOrg/torchgfn/pull/380, i.e. it doesn't allow removal of graph nodes when there is an edge associated with it by proper action masking.

It actually still learn in 300 training steps, with hyperparam adjustment.

- [x] I've read the [.github/CONTRIBUTING.md](.github/CONTRIBUTING.md) file
- [x] My code follows the typing guidelines
- [ ] I've added appropriate tests
- [x] I've run pre-commit hooks locally

### CODE REVIEW

**Review by josephdviviano — APPROVED** [review]
thanks for the fix, looks great. will you add your experimental approach in a follow up PR?
